starlite/handlers/websocket_handlers.py

```diff
@@ -99,3 +112,252 @@ def _validate_handler_function(self) -> None:
+        name: str | None = None,
+        on_accept: Callable[[WebSocket], SyncOrAsyncUnion[None]] | None = None,
+        on_disconnect: Callable[[WebSocket], SyncOrAsyncUnion[None]] | None = None,
+        opt: dict[str, Any] | None = None,
+        signature_namespace: Mapping[str, Any] | None = None,
+        type_encoders: TypeEncodersMap | None = None,
+        **kwargs: Any,
+    ) -> None:
+        """Initialize ``WebsocketRouteHandler``
+
+        Args:
+            path: A path fragment for the route handler function or a sequence of path fragments. If not given defaults
+                to ``/``
+            dependencies: A string keyed mapping of dependency :class:`Provider <.di.Provide>` instances.
+            exception_handlers: A mapping of status codes and/or exception types to handler functions.
+            guards: A sequence of :class:`Guard <.types.Guard>` callables.
+            middleware: A sequence of :class:`Middleware <.types.Middleware>`.
+            receive_mode: Websocket mode to receive data in, either `text` or `binary`.
+            send_mode: Websocket mode to receive data in, either `text` or `binary`.
+            name: A string identifying the route handler.
+            on_accept: Callback invoked after a connection has been accepted, receiving the
+                :class:`WebSocket <.connection.WebSocket>` instance as its only argument
+            on_disconnect: Callback invoked after a connection has been closed, receiving the
+                :class:`WebSocket <.connection.WebSocket>` instance as its only argument
+            opt: A string keyed mapping of arbitrary values that can be accessed in :class:`Guards <.types.Guard>` or
+                wherever you have access to :class:`Request <.connection.Request>` or
+                :class:`ASGI Scope <.types.Scope>`.
+            signature_namespace: A mapping of names to types for use in forward reference resolution during signature
+                modelling.
+            type_encoders: A mapping of types to callables that transform them into types supported for serialization.
+            **kwargs: Any additional kwarg - will be set in the opt dictionary.
+        """
+        self._receive_mode = receive_mode
+        self._send_mode = send_mode
+        self._on_accept = AsyncCallable(on_accept) if on_accept else None
+        self._on_disconnect = AsyncCallable(on_disconnect) if on_disconnect else None
+        self.type_encoders = type_encoders
+
+        super().__init__(
+            path=path,
+            dependencies=dependencies,
+            exception_handlers=exception_handlers,
+            guards=guards,
+            middleware=middleware,
+            name=name,
+            opt=opt,
+            signature_namespace=signature_namespace,
+            **kwargs,
+        )
+
+    def _validate_handler_function(self) -> None:
+        """Validate the route handler function once it's set by inspecting its return annotations."""
```

Comment:
?

Reply:
why is this method empty?

Reply:
Because the validation needs to happen elsewhere and the validation of the base classes for this needs to be disabled since it does not apply in this case.

Reply:
so can you add a comment explaining this?

Reply:
Okay so the base class of this is `WebSocketRouteHandler`, which performs a bunch of validations, all of which do not apply to this subclass. For example, it checks that handler functions return `None`, have to be asynchronous, need to have a `socket` parameter, etc. All of this is not applicable in this case, so we just disable this completely.

We cannot however use this method to validate the handler passed in by the user, because that's already been wrapped by the time `_validate_handler_function` is called, since this happens *after* it has been replaced by the handler instance; Upon being registered on the application instance.

Reply:
Does that make sense?

Reply:
i got it, i meant - add a comment in the code 😉

Reply:
Oh :cry: